test/cmd/newapp.sh

```diff
@@ -132,6 +132,21 @@ os::cmd::expect_failure_and_text 'oc new-build mysql -o yaml' 'you must specify
 os::cmd::expect_success_and_text 'oc new-build mysql --binary -o yaml --to mysql:bin' 'type: Binary'
 os::cmd::expect_success_and_text 'oc new-build mysql https://github.com/openshift/ruby-hello-world --strategy=docker -o yaml' 'type: Docker'
 os::cmd::expect_failure_and_text 'oc new-build mysql https://github.com/openshift/ruby-hello-world --binary' 'specifying binary builds and source repositories at the same time is not allowed'
+# new-build image source tests
+os::cmd::expect_failure_and_text 'oc new-build mysql --source-image centos' 'error: --source-image-path must be specified when --source-image is specified.'
+os::cmd::expect_failure_and_text 'oc new-build mysql --source-image-path foo' 'error: --source-image must be specified when --source-image-path is specified.'
```

Comment:
I think go tests run much faster and reliably than these.

Specially for testing the error messages, how about replacing these with new tests in RunBuildTests in newapp_test.go?
https://github.com/openshift/origin/pull/6615/files#diff-d47c059acde3ba8455be65d7c690ac25R1132
https://github.com/openshift/origin/pull/6615/files#diff-d47c059acde3ba8455be65d7c690ac25R1200

Those newapp_test.go tests are not the best, as they rely on GitHub, but they are easier to improve than bash.
I've just removed a lot of duplicated boilerplate from RunBuildTests.

For tests that actually require CLI interaction, then test/cmd is fine.